grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -13,6 +13,20 @@ class ProductService {
 	PermissionService permissionService
 	SubscriptionService subscriptionService
 
+	void removeUsersProducts(String username) {
+		def query = Product.where {
+			owner == username
+		}
+		def all = query.findAll()
+		all.each{ it ->
```

Comment:
`all.each { Product product ->` and can get rid of cast below

Reply:
Thats what I tried initially, but its not working as one would expect!
```
Compilation error: startup failed:
/engine-and-editor/grails-app/services/com/unifina/service/ProductService.groovy: 18: Expected parameter of type java.lang.Object but got com.unifina.domain.marketplace.Product
 @ line 18, column 13.
   		all.each{ Product product ->
               ^

/engine-and-editor/grails-app/services/com/unifina/service/ProductService.groovy: 19: [Static type checking] - Cannot find matching method java.lang.Object#delete(java.util.LinkedHashMap <java.lang.String, java.lang.Boolean>). Please check if the declared type is right and if the method exists.
 @ line 19, column 4.
   			product.delete(flush: true)
      ^

2 errors

```

Reply:
I think my grails environment is just messed

Reply:
Well both, got my grails env cleaned and setup, but still same error as above